### PR TITLE
CodeFlare Installation test

### PR DESCRIPTION
**Pull request by KPostOffice**
## How Has This Been Tested?
Will run tests on my local machine

## Merge criteria:
Depends on #21 

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: tests/basictests/distributed-workloads.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    header: "Uninstalling distributed workloads kfdef"
+}
+
+function uninstall_codeflare_operator() {
+    header: "Uninstalling Codeflare Operator"
+}
+
+
+install_codeflare_operator
+install_distributred_workloads_kfdef
```
Comment (tedhtchang) [inline tests/basictests/distributed-workloads.sh:61]:
```suggestion
install_distributed_workloads_kfdef
```

File: tests/basictests/distributed-workloads.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    # Ensure that all CRDs are created
+    os::cmd::expect_success_and_text "oc get crd instascales.codeflare.codeflare.dev  | wc -l" "2"
+    os::cmd::expect_success_and_text "oc get crd mcads.codeflare.codeflare.dev | wc -l" "2"
+    os::cmd::expect_success_and_text "oc get crd appwrappers.mcad.ibm.com | wc -l" "2"
+    os::cmd::expect_success_and_text "oc get crd queuejobs.mcad.ibm.com | wc -l" "2"
+    os::cmd::expect_success_and_text "oc get crd schedulingspecs.mcad.ibm.com | wc -l" "2"
+}
+
+function install_distributed_workloads_kfdef(){
+    header: "Installing distributed workloads kfdef"
```
Comment (tedhtchang) [inline tests/basictests/distributed-workloads.sh:29]:
```suggestion
    header "Installing distributed workloads kfdef"
```

File: tests/basictests/distributed-workloads.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    os::cmd::expect_success "oc apply -f ../distributed-workloads-kfdef.yaml"
```
Comment (tedhtchang) [inline tests/basictests/distributed-workloads.sh:30]: Where is this `distributed-workloads-kfdef.yaml` ? did you mean?
```suggestion
    os::cmd::expect_success "oc apply -f $RESOURCEDIR/../../codeflare-stack-kfdef.yaml"
```

File: tests/basictests/distributed-workloads.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    os::cmd::try_until_text "oc get role kuberay-operator-leader-election" "kuberay-operator-leader-election" $odhdefaulttimeout $odhdefaultinterval
+    os::cmd::try_until_text "oc get rolebinding kuberay-operator-leader-election" "kuberay-operator-leader-election" $odhdefaulttimeout $odhdefaultinterval
+    os::cmd::try_until_text "oc get sa kuberay-operator" "kuberay-operator" $odhdefaulttimeout $odhdefaultinterval
+    os::cmd::try_until_text "oc get deployment kuberay-operator" "kuberay-operator" $odhdefaulttimeout $odhdefaultinterval
+    os::cmd::try_until_text "oc get pods -l app.kubernetes.io/component=kuberay-operator --field-selector='status.phase=Running' -o jsonpath='{$.items[*].metadata.name}' | wc -w" "1" $odhdefaulttimeout $odhdefaultinterval   
+
+}
+
+function test_mcad_torchx_functionality() {
+    header: "Testing MCAD TorchX Functionality"
```
Comment (tedhtchang) [inline tests/basictests/distributed-workloads.sh:44]:
```suggestion
    header "Testing MCAD TorchX Functionality" 
```

File: tests/basictests/distributed-workloads.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    os::cmd::try_until_text "oc get pods -l app.kubernetes.io/component=kuberay-operator --field-selector='status.phase=Running' -o jsonpath='{$.items[*].metadata.name}' | wc -w" "1" $odhdefaulttimeout $odhdefaultinterval   
+
+}
+
+function test_mcad_torchx_functionality() {
+    header: "Testing MCAD TorchX Functionality" 
+}
+
+function tests_mcad_ray_functionality() {
+    header: "Testing MCAD Ray Functionality"
```
Comment (tedhtchang) [inline tests/basictests/distributed-workloads.sh:48]:
```suggestion
    header "Testing MCAD Ray Functionality"
```

File: tests/basictests/distributed-workloads.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+}    
+
+function uninstall_distributed_workloads_kfdef() {
+    header: "Uninstalling distributed workloads kfdef"
```
Comment (tedhtchang) [inline tests/basictests/distributed-workloads.sh:52]:
```suggestion
    header "Uninstalling distributed workloads kfdef"
```

File: tests/basictests/distributed-workloads.sh [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+function uninstall_codeflare_operator() {
+    header: "Uninstalling Codeflare Operator"
```
Comment (tedhtchang) [inline tests/basictests/distributed-workloads.sh:56]:
```suggestion
    header "Uninstalling Codeflare Operator"
```

**Comment by tedhtchang** [comment]
Tested using the following command
```
git clone https://github.com/opendatahub-io/peak
cd peak/
git submodule update --init
echo distributed-workloads nil https://github.com/KPostOffice/distributed-workloads 15-v1 > my-distributed-workloads
chmod +x operator-tests/distributed-workloads/tests/basictests/distributed-workloads.sh
./run.sh distributed-workloads
```

**Review by tedhtchang — CHANGES_REQUESTED** [review]

**Review by jbusche — APPROVED** [review]
Hi Kevin,
I just tried it and it looks good to me... CodeFlare Operator is installed as expected.

**Comment by jbusche** [comment]
Looked like a good result to me:
```
./run.sh
++++++ /root/KEVIN/peak/operator-tests/opendatahub-kubeflow/ray/tests/basictests
Now using project "basictests-bafq" on server "https://api.jimbig412.cp.fyre.ibm.com:6443".

++++ /root/KEVIN/peak/operator-tests/opendatahub-kubeflow/ray/tests/basictests/ray.sh
./run.sh: line 124: /root/KEVIN/peak/operator-tests/opendatahub-kubeflow/ray/tests/basictests/ray.sh: Permission denied
failed: /root/KEVIN/peak/operator-tests/opendatahub-kubeflow/ray/tests/basictests/ray.sh
++++++ /root/KEVIN/peak/operator-tests/opendatahub-kubeflow/tests/basictests
Already on project "basictests-bafq" on server "https://api.jimbig412.cp.fyre.ibm.com:6443".

++++ /root/KEVIN/peak/operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh

Running example test

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:15: executing 'oc project opendatahub' expecting success...


✔ SUCCESS after 0.122s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:15: executing 'oc project opendatahub' expecting success

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:16: executing 'oc get pods' expecting success...


✔ SUCCESS after 0.109s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:16: executing 'oc get pods' expecting success


Installing Codeflare Operator

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:21: executing 'oc apply -f /root/KEVIN/peak/operator-tests/opendatahub-kubeflow/tests/basictests/../resources/codeflare-subscription.yaml' expecting success...


✔ SUCCESS after 2.786s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:21: executing 'oc apply -f /root/KEVIN/peak/operator-tests/opendatahub-kubeflow/tests/basictests/../resources/codeflare-subscription.yaml' expecting success

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:24: executing 'oc get pods -n openshift-operators | grep codeflare-operator-controller-manager | awk '{print $2}'' expecting any result and text '2/2'; re-trying every 10s until completion or 1200.000s...


✔ SUCCESS after 0.122s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:24: executing 'oc get pods -n openshift-operators | grep codeflare-operator-controller-manager | awk '{print $2}'' expecting any result and text '2/2'; re-trying every 10s until completion or 1200.000s

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:27: executing 'oc get crd instascales.codeflare.codeflare.dev  | wc -l' expecting success and text '2'...


✔ SUCCESS after 0.125s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:27: executing 'oc get crd instascales.codeflare.codeflare.dev  | wc -l' expecting success and text '2'

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:28: executing 'oc get crd mcads.codeflare.codeflare.dev | wc -l' expecting success and text '2'...


✔ SUCCESS after 0.090s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:28: executing 'oc get crd mcads.codeflare.codeflare.dev | wc -l' expecting success and text '2'

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:29: executing 'oc get crd appwrappers.mcad.ibm.com | wc -l' expecting success and text '2'...


✔ SUCCESS after 0.114s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:29: executing 'oc get crd appwrappers.mcad.ibm.com | wc -l' expecting success and text '2'

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:30: executing 'oc get crd queuejobs.mcad.ibm.com | wc -l' expecting success and text '2'...


✔ SUCCESS after 0.111s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:30: executing 'oc get crd queuejobs.mcad.ibm.com | wc -l' expecting success and text '2'

Running operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:31: executing 'oc get crd schedulingspecs.mcad.ibm.com | wc -l' expecting success and text '2'...


✔ SUCCESS after 0.125s: operator-tests/opendatahub-kubeflow/tests/basictests/distributed-workloads.sh:31: executing 'oc get crd schedulingspecs.mcad.ibm.com | wc -l' expecting success and text '2'


Installing distributed workloads kfdef


Testing MCAD TorchX Functionality


Testing MCAD Ray Functionality


Uninstalling distributed workloads kfdef


Uninstalling Codeflare Operator

Already on project "opendatahub" on server "https://api.jimbig412.cp.fyre.ibm.com:6443".
One or more tests failed:

	/root/KEVIN/peak/operator-tests/opendatahub-kubeflow/ray/tests/basictests/ray.sh

./run.sh took 5 seconds
[INFO] Exiting with 1
```
and
```
oc get pods -n openshift-operators
NAME                                                    READY   STATUS    RESTARTS   AGE
codeflare-operator-controller-manager-6579fb6c5-4b97p   2/2     Running   0          53s
opendatahub-operator-66dd4b8fbc-tzjm6                   1/1     Running   0          24m
```

**Comment by KPostOffice** [comment]
@jbusche Do you have write access to approve this?

**Review by jbusche — APPROVED** [review]
lgtm

**Comment by jbusche** [comment]
Hi @KPostOffice, no - just tried approving again, and it's not seeming to allow my approve.  I MIGHT have merge capability once the approval is done, but I suspect not.

**Review by MichaelClifford — APPROVED** [review]

**Pull request merged**